nixos/modules/services/hardware/throttled.nix

```diff
@@ -0,0 +1,19 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.throttled;
+in {
+  options = {
+    services.throttled = {
+      enable = mkEnableOption "fix for Intel CPU throttling";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.throttled ];
```

Comment:
Should the module not also enable the service automatically or how is this supposed to be used?

Reply:
It's got `WantedBy=multi-user.target`, so just installing the unit seems to be enough.

Reply:
@Mic92 Are you happy with this explanation?

Reply:
This will not make it automatically started:

```
$ systemctl status lenovo_fix
● lenovo_fix.service - Stop Intel throttling
   Loaded: loaded (/nix/store/a0zg97bwn74k2mn7wxhzibrf3d3rbnqq-throttled-0.6/lib/systemd/system/lenovo_fix.service; linked; vendor preset: enabled)
   Active: inactive (dead)
```

We need to explicitly have a `wantedBy` attribute set in nix code so it will started on bootup.

Reply:
Hm, I'm confused:
```
michael@clipper:~/projects/forks/nixpkgs ❯ systemctl cat lenovo_fix                                                      fix/redshift-geoclue | clean
# /nix/store/63kmhns86s282p3h46nif3agcrykl2j6-throttled-0.6/lib/systemd/system/lenovo_fix.service
[Unit]
Description=Stop Intel throttling 

[Service]
Type=simple
ExecStart=/nix/store/63kmhns86s282p3h46nif3agcrykl2j6-throttled-0.6/bin/lenovo_fix.py
# Setting PYTHONUNBUFFERED is necessary to see the output of this service in the journal
Environment=PYTHONUNBUFFERED=1
StandardOutput=syslog
StandardError=syslog

[Install]
WantedBy=multi-user.target
```
It looks like it already has `WantedBy`? And I'm pretty sure I didn't explicitly enable it but it's started for me.

Reply:
I read the nixos manual. You're right - I fixed it.